nullaway/src/main/java/com/uber/nullaway/dataflow/cfg/NullAwayCFGBuilder.java

```diff
@@ -0,0 +1,183 @@
+   * @param bodyPath the TreePath to the body of the method, lambda, or initializer.
+   * @param underlyingAST the AST that underlies the control frow graph
+   * @param assumeAssertionsEnabled can assertions be assumed to be disabled?
+   * @param assumeAssertionsDisabled can assertions be assumed to be enabled?
+   * @param env annotation processing environment containing type utilities
+   * @param handler a NullAway handler or chain of handlers (through {@link
+   *     com.uber.nullaway.handlers.CompositeHandler}
+   * @return a control flow graph
+   */
+  public static ControlFlowGraph build(
+      TreePath bodyPath,
+      UnderlyingAST underlyingAST,
+      boolean assumeAssertionsEnabled,
+      boolean assumeAssertionsDisabled,
+      ProcessingEnvironment env,
+      Handler handler) {
+    TreeBuilder builder = new TreeBuilder(env);
+    AnnotationProvider annotationProvider = new BasicAnnotationProvider();
+    CFGTranslationPhaseOne phase1translator =
+        new NullAwayCFGTranslationPhaseOne(
+            builder,
+            annotationProvider,
+            assumeAssertionsEnabled,
+            assumeAssertionsDisabled,
+            env,
+            handler);
+    PhaseOneResult phase1result = phase1translator.process(bodyPath, underlyingAST);
+    ControlFlowGraph phase2result = CFGTranslationPhaseTwo.process(phase1result);
+    ControlFlowGraph phase3result = CFGTranslationPhaseThree.process(phase2result);
+    return phase3result;
+  }
+
+  /**
+   * A NullAway specific subclass of the Checker Framework's {@link CFGTranslationPhaseOne},
+   * augmented with handler extension points and some utility methods meant to be called from
+   * handlers to customize the AST to CFG translation.
+   */
+  public static class NullAwayCFGTranslationPhaseOne extends CFGTranslationPhaseOne {
+
+    private final Handler handler;
+
+    /**
+     * Create a new NullAway phase one translation visitor.
+     *
+     * @param builder a TreeBuilder object (used to create synthetic AST nodes to feed to the
+     *     translation process)
+     * @param annotationProvider an {@link AnnotationProvider}.
+     * @param assumeAssertionsEnabled can assertions be assumed to be disabled?
+     * @param assumeAssertionsDisabled can assertions be assumed to be enabled?
+     * @param env annotation processing environment containing type utilities
+     * @param handler a NullAway handler or chain of handlers (through {@link
+     *     com.uber.nullaway.handlers.CompositeHandler}
+     */
+    public NullAwayCFGTranslationPhaseOne(
+        TreeBuilder builder,
+        AnnotationProvider annotationProvider,
+        boolean assumeAssertionsEnabled,
+        boolean assumeAssertionsDisabled,
+        ProcessingEnvironment env,
+        Handler handler) {
+      super(builder, annotationProvider, assumeAssertionsEnabled, assumeAssertionsDisabled, env);
+      this.handler = handler;
+    }
+
+    @SuppressWarnings("NullAway") // (Void)null issue
```

Comment:
This is a NullAway bug, I would think, but arguably outside the scope of this PR.

Reply:
Weird.  We should file a separate issue on this.

Reply:
Tracking this as #611